content/docs/concepts/resources/functions.md

```diff
@@ -131,34 +131,195 @@ variables:
+{
+    var project = new Gcp.Organizations.Project("project", new()
+    {
+        Name = "project",
+        ProjectId = "project",
+        BillingAccount = "bills",
+        AutoCreateNetwork = false,
+    });
+
+    // Since `project.Name` is immediately available, we need to use `Output.All`
+    // and `Output.Apply` to ensure that `project` is created and available before
+    // calling `Gcp.Compute.GetZonesOutput`.
+    Output.Tuple(project.Name, project.Urn)
+        .Apply(t => Gcp.Compute.GetZonesOutput.Invoke(new Gcp.Compute.GetZonesOutputArgs
+        {
+            Region = "us-central1",
+            Project = args.Item1,
+            Status = "UP",
+        }));
+});
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="go">
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi-gcp/sdk/v7/go/gcp/compute"
+	"github.com/pulumi/pulumi-gcp/sdk/v7/go/gcp/organizations"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		project, err := organizations.NewProject(ctx, "project", &organizations.ProjectArgs{
+			Name:              pulumi.String("project"),
+			ProjectId:         pulumi.String("project"),
+			BillingAccount:    pulumi.String("bills"),
+			AutoCreateNetwork: false,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Since `project.Name` is immediately available, we need to use `pulumi.All`
+		// and `pulumi.ApplyT` to ensure that `project` is created and available before
+		// calling `compute.GetZonesOutput`.
+		_, err = pulumi.All(project.Name, project.Urn).ApplyT(
+			func(args []interface{}) (interface{}, error) {
+				project := args[0].(string)
+				return compute.GetZonesOutput(ctx, &compute.GetZonesOutputArgs{
+					Region:  pulumi.String("us-central1"),
+					Project: pulumi.String(project),
+					Status:  pulumi.String("UP"),
+				})
+			},
+		)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="python">
+
+```python
+import pulumi
+import pulumi_gcp as gcp
+
+project = gcp.organizations.Project("project",
+    name="project",
+    project_id="project",
+    billing_account="bills",
+    auto_create_network=False)
+
+# Since `project.name` is immediately available, we need to use `pulumi.Output.all`
+# and `pulumi.Output.apply` to ensure that `project` is created and available before
+# calling `gcp.compute.get_zones`.
+pulumi.Output.all(project.name, project.urn) \
+    .apply(lambda args: gcp.compute.get_zones_output(
+        region="us-central1",
+        project=args[0],
+        status="UP"
+    ))
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="java">
+
+Output form invocations are not yet supported in Java
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="yaml">
+
+Output form invocations are not yet supported in YAML
```

Comment:
```suggestion
Output form invocations are not supported in YAML
```

I'm not sure we'll ever want to support them?  "Not yet" gives the impression that we do want to.

Reply:
I don't see why we couldn't support them in YAML eventually?

Reply:
Fair, I guess I didn't see how they fit in well with yaml, but I also didn't think long about it.